### PR TITLE
feat: filter NFT spammy domains

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

This PR filters NFTs which have spammy domains medias, to avoid encountering the Chrome red security screen when images are fetched for NFTs/assets.

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

contributes to the closed https://github.com/shapeshift/web/issues/5301

## Risk

None

## Testing

- Apply the diff below to disable the current filtering of spammy NFTs

```diff
diff --git a/src/state/apis/nft/constants.ts b/src/state/apis/nft/constants.ts
index 85d1454752..a7e3eb8170 100644
--- a/src/state/apis/nft/constants.ts
+++ b/src/state/apis/nft/constants.ts
@@ -4,26 +4,7 @@ import type { NftItem } from './types'
 
 // Try to be as broad as possible vs. specific here to accommodate for diff. TLDs and subdomains
 const NFT_DOMAINS_BLACKLIST = ['ethercb']
-const NFT_NAME_BLACKLIST = [
-  'voucher',
-  'airdrop',
-  'giveaway',
-  'promo',
-  'airdrop',
-  'rewards',
-  'ticket',
-  'winner',
-  '$',
-  'pirategirls',
-  'USDC',
-  'USDT',
-  'calim cryptopunk',
-  'coupon',
-  'bonus',
-  'claim',
-  'etherscan',
-  'shibarium',
-]
+const NFT_NAME_BLACKLIST = ['nothingtoseehere']
 
 // This escapes special characters we may encounter in NFTS, so we can add them to the blacklist
 // e.g "$9999+ free giveaway *limited time only*" would not work without it
@@ -36,25 +17,4 @@ const isSpammyDomain = (domain: string) =>
   NFT_DOMAINS_BLACKLIST.some(blacklistedDomain => domain.includes(blacklistedDomain))
 export const hasSpammyMedias = (medias: NftItem['medias']) =>
   medias.some(media => isSpammyDomain(media.originalUrl))
-export const BLACKLISTED_COLLECTION_IDS = [
-  'eip155:137/erc1155:0x30825b65e775678997c7fbc5831ab492c697448e',
-  'eip155:137/erc1155:0x4217495f2a128da8d6122d120a1657753823721a',
-  'eip155:137/erc1155:0x54e75b47353a5ded078d8eb6ba67ff01a2a18ef7',
-  'eip155:137/erc1155:0x7d33048b74c24d695d349e73a59b7de3ed50c4c0',
-  'eip155:137/erc1155:0xb1cfea0eb0f67a50b896985bda6368fc1c21907b',
-  'eip155:137/erc1155:0xc165d899628c5fd74e19c91de596a9ea2f3599ec',
-  'eip155:137/erc1155:0xcf2576238640a3a232fa6046d549dfb753a805f4',
-  'eip155:137/erc1155:0xcf63b89da7c6ada007fbef13fa1e8453756ba7a6',
-  'eip155:137/erc1155:0xda8091a96aefcc2bec5ed64eb2e18008ebf7806c',
-  'eip155:137/erc1155:0xe0b7dafe2eb86ad56386676a61781d863144db1e',
-  'eip155:137/erc1155:0xed13387ea51efb26b1281bd6decc352141fd312a',
-  'eip155:137/erc1155:0xf02521228c6250b255abbc4ea66fd5aa86aa2ce0',
-  'eip155:137/erc1155:0xf30437ad7d081046b4931e29460fcb0d7bbaca46',
-  'eip155:137/erc1155:0xfd920bd499511d0f5e37b4405a7986a4d6f1abe3',
-  'eip155:137/erc1155:0x55d50a035bc5830dac9f1a42b71c48cbad568d60',
-  'eip155:137/erc1155:0x5620a667cbe1eb7e1e27087d135881a546456ebb',
-  'eip155:137/erc1155:0x1b6ff968f954fef75a82c0f47364eb00c3643d17',
-  'eip155:137/erc1155:0x5df82d6f8dc0a697d2785953a40735b33e3b1f50',
-  'eip155:137/erc1155:0x71c7c713aaf6b129ebeb29d00647b6cc39e4f9f9',
-  'eip155:137/erc1155:0xabda1f46fe8ad6b08aa9505702144522c511038f',
-]
+export const BLACKLISTED_COLLECTION_IDS = []
```

- Ensure the $ETH NFT is filtered and we're not hitting the scammy domain

### Engineering

- ☝🏽 

### Operations

- Ensure you still can't see the $ETH scammy NFT that's causing red chrome

## Screenshots (if applicable)

- With the diff above applied, and without this PR diff

Screenshot 2023-09-19 at 14 19 04
Screenshot 2023-09-19 at 14 18 57

- With this diff, as well as the diff above applied

Screenshot 2023-09-19 at 14 23 14
Screenshot 2023-09-19 at 14 22 51

### CODE REVIEW

File: src/state/apis/nft/nftApi.ts [hunk 1]
```diff
@@ -266,6 +263,7 @@ export const nftApi = createApi({
             if (!item.collection.assetId) return acc
             const cachedCollection = selectNftCollectionById(state, item.collection.assetId)
             if (cachedCollection?.isSpam) item.collection.isSpam = true
+            if (hasSpammyMedias(item.medias)) item.collection.isSpam = true
```
Comment (kaladinlight) [inline src/state/apis/nft/nftApi.ts:266]: we are already filtering spammy medias out above

File: src/state/apis/nft/nftApi.ts [hunk 2]
```diff
@@ -266,6 +263,7 @@ export const nftApi = createApi({
...
             if ([item.description, item.name, item.symbol].some(isSpammyNftText))
```
Comment (kaladinlight) [inline src/state/apis/nft/nftApi.ts:267]: ditto

File: src/state/apis/nft/nftApi.ts (hunk 1 above)
Comment (gomesalexandre) [reply src/state/apis/nft/nftApi.ts]: Nice catch, that's def. a redundant filtering fe4e5a1d06f9e7f0582813c14d59e535724666d1

File: src/state/apis/nft/nftApi.ts [hunk 3]
```diff
@@ -266,6 +263,7 @@ export const nftApi = createApi({
             if (!item.collection.assetId) return acc
             const cachedCollection = selectNftCollectionById(state, item.collection.assetId)
             if (cachedCollection?.isSpam) item.collection.isSpam = true
```
[the remark this answers is not in the document]
Comment (gomesalexandre) [reply src/state/apis/nft/nftApi.ts]: fe4e5a1d06f9e7f0582813c14d59e535724666d1

File: src/state/apis/nft/nftApi.ts (hunk 2 above)
Comment (gomesalexandre) [reply src/state/apis/nft/nftApi.ts]: fe4e5a1d06f9e7f0582813c14d59e535724666d1

**Review by kaladinlight — APPROVED** [review]

**Pull request merged**